components/input/textarea-count.component.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
+ */
+import { AfterContentInit, ChangeDetectionStrategy, Component, ContentChild, ElementRef, Input, OnDestroy, Renderer2 } from '@angular/core';
+import { isNotNil } from 'ng-zorro-antd/core/util';
+import { NzInputDirective } from 'ng-zorro-antd/input/input.directive';
+import { EMPTY, Subject } from 'rxjs';
+import { map, startWith, takeUntil } from 'rxjs/operators';
+
+@Component({
+  selector: 'nz-textarea-count',
+  template: `
+    <ng-content select="textarea[nz-input]"></ng-content>
+  `,
+  host: {
+    class: 'ant-input-textarea-show-count'
+  },
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class NzTextareaCountComponent implements AfterContentInit, OnDestroy {
+  @ContentChild(NzInputDirective, { static: true }) nzInputDirective!: NzInputDirective;
```

Comment:
This doesn't need to be static: true, does it?

Reply:
thx review, dynamic content will make matching targets more complex, in which case I think it's enough for developers.

Reply:
The static flag isn't about dynamic content, it's about lifecycle, and static true is deprecated so it has to be removed anyway. In here it can be removed without any additional effort.